staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured_test.go

```diff
@@ -16,7 +16,31 @@ limitations under the License.
 
 package unstructured
 
-import "testing"
+import (
+	"io/ioutil"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestCodecOfUnstructuredList tests that there are no data races in Encode().
+// i.e. that it does not mutate the object being encoded.
+func TestCodecOfUnstructuredList(t *testing.T) {
```

Comment:
I wonder if this test fails without your change or will it be flaky. Checking race condition is hard and I am not sure if `go build --race` can be helpful here or not. but we can at least test to make sure a value with 'items' key in that map would not be removed, right?

Reply:
wait. -race is enabled for our test. so this test is fine.